### PR TITLE
[Custom/Loss] Add scaffolding for loss example

### PR DESCRIPTION
**Pull request by zhoonit**
- [Custom/Loss] Add scaffolding for loss example

```
This patch generates a skeleton code for mae custom loss example.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

**Review by lhs8928 — APPROVED** [review]
LGTM

File: Applications/Custom/mae_loss.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   mae_loss.cpp
+ * @date   10 June 2021
+ * @brief  This file contains the mean absoulte error loss as a sample layer
```
Comment (kparichay) [inline Applications/Custom/mae_loss.cpp:7]: absolute*

File: Applications/Custom/mae_loss.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ */
+#include "mae_loss.h"
+
+namespace custom {
+const std::string MaeLossLayer::type = "mae_loss";
+
+#ifdef PLUGGABLE
+
+nntrainer::Layer *create_mae_loss_layer() {
```
Comment (kparichay) [inline Applications/Custom/mae_loss.cpp:20]: this would update to LayerV1?

File: Applications/Custom/mae_loss.h [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+/// @todo migrate these to API and ensure those headers are exposed to devel
+#include <layer_internal.h>
+#include <manager.h>
+#include <tensor.h>
+
+namespace custom {
+
+/**
+ * @brief A sample loss layer which calculates mean absolute error from output
+ *
```
Comment (kparichay) [inline Applications/Custom/mae_loss.h:26]: add a todo to update inheritance from layer_devel?

File: Applications/Custom/pow.h [hunk 4]
```diff
@@ -15,17 +15,21 @@
...
 
-/// @todo inherit this to API
-// class PowLayer : public ml::train::Layer {
-class PowLayer : public nntrainer::Layer {
+/**
+ * @brief layer class that calculates f(x) = x ^ exponent (exponent is
+ * configurable by PowLayer::setProperty)
+ *
+ */
+class PowLayer final : public nntrainer::Layer {
```
Comment (kparichay) [inline Applications/Custom/pow.h:32]: Layer -> LayerV1?

**Review by kparichay — APPROVED** [review]
LGTM other than the minor comments

File: Applications/Custom/mae_loss.cpp (hunk 1 above)
Comment (zhoonit) [reply Applications/Custom/mae_loss.cpp:7]: Done

File: Applications/Custom/mae_loss.cpp (hunk 2 above)
Comment (zhoonit) [reply Applications/Custom/mae_loss.cpp:20]: Done

File: Applications/Custom/mae_loss.h (hunk 3 above)
Comment (zhoonit) [reply Applications/Custom/mae_loss.h:26]: Done

File: Applications/Custom/pow.h (hunk 4 above)
Comment (zhoonit) [reply Applications/Custom/pow.h]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: Applications/Custom/mae_loss.h [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  int setProperty(std::vector<std::string> values) override { return 0; }
+
+  /**
+   * @brief initializing nntrainer
+   *
+   * @return int ML_ERROR_NONE if success
+   */
+  int initialize(nntrainer::Manager &manager) override { return 0; }
```
Comment (jijoongmoon) [inline Applications/Custom/mae_loss.h:58]: it might be mark NYI with return 0;

Comment (zhoonit) [reply Applications/Custom/mae_loss.h]: Done

**Pull request merged**